python-avd/pyavd/_eos_designs/shared_utils/link_tracking_groups.py

```diff
@@ -21,18 +22,17 @@ class LinkTrackingGroupsMixin(Protocol):
     """
 
     @cached_property
-    def link_tracking_groups(self: SharedUtilsProtocol) -> list | None:
+    def link_tracking_groups(self: SharedUtilsProtocol) -> EosCliConfigGen.LinkTrackingGroups | None:
         if self.node_config.link_tracking.enabled:
-            link_tracking_groups = []
+            link_tracking_groups = EosCliConfigGen.LinkTrackingGroups()
             default_recovery_delay = default(self.platform_settings.reload_delay.mlag, 300)
             if len(self.node_config.link_tracking.groups) > 0:
                 for lt_group in self.node_config.link_tracking.groups:
-                    lt_group_dict = lt_group._as_dict(include_default_values=True)
-                    lt_group_dict["recovery_delay"] = default(lt_group.recovery_delay, default_recovery_delay)
-                    link_tracking_groups.append(lt_group_dict)
+                    lt_group_item = lt_group._cast_as(EosCliConfigGen.LinkTrackingGroupsItem, ignore_extra_keys=True)
+                    lt_group_item.recovery_delay = default(lt_group.recovery_delay, default_recovery_delay)
+                    link_tracking_groups.append(lt_group_item)
```

Comment:
There are only two keys in the link tracking group so I would prefer to avoid the cast_as since we also override the recovery_delay anyway.
```suggestion
                    link_tracking_groups.append_new(
                        name=lt_group.name,
                        links_minimum=lt_group.links_minimum,
                        recovery_delay=default(name=lt_group.recovery_delay, default_recovery_delay),
                    )
```